guides/managing-current-user.md

```diff
@@ -146,13 +146,14 @@ const { service } = Ember.inject;
 export default Ember.Route.extend(ApplicationRouteMixin, {
   currentUser: service(),
 
-  beforeModel() {
-    return this._loadCurrentUser();
+  async beforeModel() {
+    await this._loadCurrentUser();
+    return this._super(...arguments);
   },
 
-  sessionAuthenticated() {
-    this._super(...arguments);
-    this._loadCurrentUser();
+  async sessionAuthenticated() {
+    await this._loadCurrentUser();
+    return this._super(...arguments);
```

Comment:
I think the `return` is misleading as `_super` doesn't return anything and you'd probably want to call `_super` first.

Reply:
I copied this from your PR comment. :) but looking at the source code, I agree the `return` is unnecessary. Calling `_super` first is actually the bug this is trying to fix. You need to wait for the user to load before you have the mixin process whatever transition is currently happening.

Reply:
I also see in the [`beforeModel`](https://api.emberjs.com/ember/3.10/classes/Route/methods/beforeModel?anchor=beforeModel) docs that it will already pause transitions when a Promise is returned, so I think the current documentation was fine there.

Reply:
Ah, sorry - you're right of course 🤦‍♂